lua/roslyn/config.lua

```diff
@@ -1,7 +1,7 @@
 local M = {}
 
 ---@class InternalRoslynNvimConfig
----@field filewatching boolean
+---@field filewatching string values: auto, force-nvim, force-ls
```

Comment:
```suggestion
---@field filewatching "auto" | "roslyn" | "off"
```

I am just using `off` for now, but you get the gist

Reply:
You also need to change the `RoslynNvimConfig` class.

This should have the type:
```lua
---@field filewatching? boolean | "auto" | "off" | "roslyn"
```
since otherwise lua_ls will complain further down I think

Reply:
Yes, at the beginning I also named these options "auto" | "off" | "roslyn", but then changed my mind. It's probably worth going back to these names. I'll try to fix everything tomorrow.

Reply:
> You also need to change the `RoslynNvimConfig` class.
> 
> This should have the type:
> 
> ```lua
> ---@field filewatching? boolean | "auto" | "off" | "roslyn"
> ```
> 
> since otherwise lua_ls will complain further down I think

Maybe it would be better to remove "boolean"?